### DISCUSSION TITLE
How to specify the size of the page when transforms into PCL file

### DISCUSSION THREAD

**Discussion by avasquez-ve** [dlemstra/Magick.NET#1370] [Help]
Hi everyone!

Currently i'm trying to transform image in bmp format into PCL files,

The transformation works and I could print the image using the font format of PCL,

My problem is:

- All the images are transformed in letter page size and I can't find a way to specify a custome page size, for example 2x5 inches. That generate my image is always printed in a different page of my main content.

My question is:
- How can I set the size of the page/canvas of my PCL file when is transformed?

I'm using the following piece of code to transform and try to specify the size, but the result always is a complete page.

```
using (var image = new MagickImage(@"C:\myPath\imagen16colores.bmp"))
{
	image.Format = MagickFormat.Pcl;
	image.Page = new MagickGeometry(1000, 50, 100, 200);
	image.Write(@"C:\myPath\outputFile.pcl");
}
```

The result is:
My transformed image into a whole page (I'm using CoolUtils PCL Viewer to se the result, but when I printed the PCL file using PCL commands I got the same result)
![image](https://user-images.githubusercontent.com/24895868/229572904-949771aa-7c4b-4a37-b213-ed7c8ce63772.png)
![image](https://user-images.githubusercontent.com/24895868/229572988-b6584d02-d283-4802-b283-b73039f2a47d.png)

**Comment by dlemstra** [comment]
What happens when you do `image.Settings.SetDefine(MagickFormat.Pcl, "fit-to-page", false);` instead of setting the Page?

p.s. If this works I will add PclWriteDefines in the next release to make this easier for you.

**Reply by avasquez-ve** [reply]
Hi Dirk!

Did you mean writting the following lines?

```
using (var image = new MagickImage(@"C:\myPath\imagen16colores.bmp"))
{
	image.Settings.SetDefine(MagickFormat.Pcl, "fit-to-page", false);

	image.Write(@"C:\myPath\output.pcl");
}
```

By using the PCL Viewer I got the same output, but let me execute the printing in a while,

**Reply by dlemstra** [reply]
That's what I meant. Have you tried printing it?

**Reply by avasquez-ve** [reply]
Hi Dirk, sorry for the delay,

I tried by using the option you gave me, but the result is the same: The image was printed as a new page.

It's weird becase I can resize the image, but I can't do the same to specify the size of the page/canvas.

**Reply by avasquez-ve** [reply]
Dirk,

I found a way to display the image in the same page of the current content, 

Tomorrow I will execute the printing test to validate if is also recognized,

If is success, the fact would be related to how PCL functions rather than the transformation process with Magick.Net..

Until then, I'll let you know

**Reply by dlemstra** [reply]
Would like to know how you fixed it because it sounds like we are writing the PCL file incorrect in ImageMagick then? So please share how you fixed it.
